conda_smithy/schema.py

```diff
@@ -0,0 +1,1217 @@
+        default=16,
+        description="""The compression level for the zstd compression algorithm for
+            .conda artifacts. conda-forge uses a default value of 16 for a good
+            compromise of performance and compression.""",
+    )
+
+    error_overlinking: Optional[bool] = Field(
+        default=False,
+        description="""
+            Enable error when shared libraries from transitive dependencies are
+            directly  linked  to any executables or shared libraries in  built
+            packages. For more details, see the
+            [conda build documentation](https://docs.conda.io/projects/conda-build/en/stable/resources/commands/conda-build.html).
+            """,
+    )
+
+
+class CondaForgeDocker(BaseModel):
+    executable: Optional[str] = Field(
+        description="The executable for Docker", default="docker"
+    )
+
+    fallback_image: Optional[str] = Field(
+        description="The fallback image for Docker",
+        default="quay.io/condaforge/linux-anvil-comp7",
+    )
+
+    command: Optional[str] = Field(
+        description="The command to run in Docker", default="bash"
+    )
+
+    interactive: Optional[Union[bool, Nullable]] = Field(
+        description="Whether to run Docker in interactive mode",
+        default=None,
+        exclude=True,  # Will not be rendered in the model dump
+    )
+
+    #########################################
+    #### Deprecated Docker configuration ####
+    #########################################
+    image: Optional[Union[str, Nullable]] = Field(
+        description="""Setting the Docker image in conda-forge.yml is no longer
+        supported, use conda_build_config.yaml to specify Docker images.""",
+        default=None,
+        exclude=True,  # Will not be rendered in the model dump
+    )
+
+
+class ShellCheck(BaseModel):
+    enabled: bool = Field(
+        description="Whether to use shellcheck to lint shell scripts",
+        default=False,
+    )
+
+
+class PlatformsAliases(StrEnum):
+    linux = "linux"
+    win = "win"
+    osx = "osx"
+
+
+def get_subdirs():
+    return [
+        subdir.replace("-", "_") for subdir in KNOWN_SUBDIRS if "-" in subdir
+    ]
+
+
+Platforms = StrEnum("Platforms", get_subdirs())
+
+
+class ChannelPriorityConfig(str, Enum):
+    STRICT = "strict"
+    FLEXIBLE = "flexible"
+    DISABLED = "disabled"
+
+
+class DefaultTestPlatforms(str, Enum):
+    all = "all"
+    native_only = "native_only"
+    native_and_emulated = "native_and_emulated"
+    emulated_only = "emulated_only"
+
+
+class ConfigModel(BaseModel):
+    """
+    This model describes in detail the top-level fields in  ``conda-forge.yml``.
+    General configuration options are described below within the ``Fields``
+    specifications. Additional examples are provided as part of the object
+    description. Values and options are subject to change, and will be
+    flagged as Deprecated as appropriate.
+
+    """
+
```

Comment:
Recommend adding some metadata which will make it out to the `.json`.

```py
class ConfigModel(BaseModel):
    class Config:
        schema_extra = {
            "$schema": "https://json-schema.org/draft/2020-12/schema#",
            "$id": "https://conda-forge.org/conda-smithy/conda-forge.v2.schema.json",
        }
```

Reply:
While, _technically_, `$id` is just an URI (identifier) and doesn't need to be locate-able. However, it is _really good_ if they are, once the time is taken to actually build them. A top-level subdirectory is a fine idea. This could hold other things, too, like:
- `repodata-patches` minigrammar
- `county-fair` content descriptions

Indeed, if the schema permitted `$schema`, and a user has a [YAML language server](https://github.com/redhat-developer/yaml-language-server) loaded up in their editor of choice, and something like:

```yaml
$schema: https://conda-forge.org/conda-smithy/conda-forge.v2.schema.json
bo # press <tab>
```

They might well see `bot`, and some helpful documentation. I haven't _yet_ wired all of these up into a statically-hostable docs "playground" site, but have been slowly making progress towards it.

Reply:
Let's add that when there is a PR to the website.